WcaOnRails/app/controllers/api/v0/persons_controller.rb

```diff
@@ -0,0 +1,38 @@
+# frozen_string_literal: true
+
+class Api::V0::PersonsController < Api::V0::ApiController
+  def index
+    persons = Person.current.includes(:user, :ranksSingle, :ranksAverage)
+    if params[:wca_ids].present?
+      persons = persons.where(wca_id: params[:wca_ids].split(','))
+    end
+    render json: paginate(persons).map { |person| person_to_json person }
+  end
+
+  def show
+    person = Person.current.includes(:user, :ranksSingle, :ranksAverage).find_by_wca_id!(params[:wca_id])
+    render json: person_to_json(person)
+  end
+
+  private def person_to_json(person)
+    {
+      person: person.serializable_hash.slice(:wca_id, :name, :url, :gender, :country_iso2, :delegate_status, :teams, :avatar),
+      personal_records: person.ranksSingle.each_with_object({}) do |rank_single, ranks|
+        event_id = rank_single.event.id
+        rank_average = person.ranksAverage.find { |rank| rank.event_id == event_id }
+        ranks[event_id] = { single: rank_to_json(rank_single) }
+        ranks[event_id][:average] = rank_to_json(rank_average) if rank_average
+      end,
+    }
+  end
+
+  private def rank_to_json(rank)
+    {
+      best: rank.best,
+      best_formatted: rank.solve_time.clock_format,
```

Comment:
I'm still torn about this. Can you think of an example consumer of this API who would benefit from this? My current thinking is that most people using this API would need to write the code to parse `best` anyways, and they very possibly will have to write rendering code anyways as well.

I'm thinking maybe a better thing we could do would be to provide libraries for common languages to handle this sort of stuff, but I realize that's a bit of work...

@viroulep, @coder13, @larspetrus what do you think?

Reply:
As said offline, I put it here just because I felt someone might want to display it easily without having to write the whole parsing code. If you think otherwise I will remove it, we can always add it back if someone requests this.

Reply:
So `best` is a number and `best_formatted` is it  nicely formatted?

If the formatting is just 17.2 -> "17.20", it doesn't seem worth it. The more advanced it is, the more attractive it gets.

Reply:
@larspetrus even for standard values we store centiseconds, so it does make a difference, but it's more about formatting minutes, hours and complex things like multibld.

Reply:
I'd rather take the `*_formatted` out, and make clear that we have a reference implementation for that [here](https://github.com/thewca/worldcubeassociation.org/blob/master/WcaOnRails/lib/solve_time.rb).

Ideally we could make this some small gem, and we would provide another implementation in js for example, but I don't think it's a big challenge for someone to adapt the ruby code to another language.

Reply:
Right, so everyone seems convinced that it doesn't make much sense, at least at this point. I'm going to remove it and we can always add it if we find out it is needed =)

Reply:
I don't mind that we wait to see if people want this, but just for the record, *I* think it's obviously useful for any user who intend to display a formatted result. If that's a common use case, it's probably a worthwhile feature.

Providing code to compute the string in various languages is obviously inferior to simply providing the actual string.